src/actions/vdom/add-attribute.js

```diff
@@ -0,0 +1,16 @@
+var toCamelCase = require('../../utils/to-camel-case');
+
+module.exports = function addAttribute(node, key, val) {
+
+  node = (node.vNode || node);
+  node.properties.dataset = (node.properties.dataset || {});
+
+  if (/data/.test(key)) {
```

Comment:
This regexp will match "database", "metadata", etc. Maybe you want `/^data-.+/` instead?